theano/tensor/nnet/abstract_conv.py

```diff
@@ -216,24 +217,50 @@ def conv2d_grad_wrt_inputs(output_grad,
 
     """
 
-    grad_input_op = AbstractConv2d_gradInputs(imshp=input_shape,
+    filters = as_tensor_variable(filters)
+    output_grad = as_tensor_variable(output_grad)
+
+    # checking the type of input_shape
+    for dim in [0, 1]:
+        assert isinstance(input_shape[dim], (theano.tensor.TensorConstant,
+                                             int, type(None)))
+    for dim in [2, 3]:
+        assert isinstance(input_shape[dim], (theano.tensor.TensorVariable,
+                                             theano.tensor.TensorConstant,
+                                             int))
+
+    # checking the type of filter_shape
+    for dim in [0, 1, 2, 3]:
+        assert isinstance(filter_shape[dim], (theano.tensor.TensorVariable,
```

Comment:
Just for reference, as we discussed IRL a few minutes ago `filter_shape` should be either `None` or a 4-elements Iterable whose elements can be either `None`, `int` or `Constant`.